src/Sylius/Behat/Context/Api/Admin/ManagingShippingCategoriesContext.php

```diff
@@ -0,0 +1,220 @@
+     */
+    public function iShouldBeNotifiedThatElementIsRequired(string $element): void
+    {
+        Assert::same(
+            $this->client->getError(), sprintf('%s: Please enter shipping category %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then I should see a single shipping category in the list
+     * @Then I should see :count shipping categories in the list
+     */
+    public function iShouldSeeShippingCategoriesInTheList(int $count = 1): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::same($this->client->countCollectionItems(), $count);
+    }
+
+    /**
+     * @Then /^the (shipping category "([^"]+)") should be in the registry$/
+     * @Then /^the (shipping category "([^"]+)") should appear in the registry$/
+     */
+    public function theShippingCategoryShouldAppearInTheRegistry(ShippingCategoryInterface $shippingCategory): void
+    {
+        $shippingCategoryName = $shippingCategory->getName();
+        Assert::true(
+            $this->isItemOnIndex('name', $shippingCategoryName),
+            sprintf('Shipping category with name %s does not exists', $shippingCategoryName)
+        );
+    }
+
+    /**
+     * @Then shipping category with name :name should not be added
+     */
+    public function shippingCategoryWithNameShouldNotBeAdded(string $name): void
+    {
+        Assert::false(
+            $this->isItemOnIndex('name', $name),
+            sprintf('Shipping category with name %s exists', $name)
+        );
+    }
+
+    /**
+     * @Then /^(this shipping category) should no longer exist in the registry$/
+     */
+    public function thisShippingCategoryShouldNoLongerExistInTheRegistry(
+        ShippingCategoryInterface $shippingCategory
+    ): void {
+        $shippingCategoryName = $shippingCategory->getName();
+        Assert::false(
+            $this->isItemOnIndex('name', $shippingCategoryName),
+            sprintf('Shipping category with name %s exist', $shippingCategoryName)
+        );
+    }
+
+    /**
+     * @Then I should not be able to edit its code
+     */
+    public function iShouldNotBeAbleToEditItsCode(): void
+    {
+        $this->client->addRequestData('code', 'NEW_CODE');
+        $this->client->update();
+
+        Assert::false($this->client->hasValue('code', 'NEW_CODE'), 'The code field with value NEW_CODE exists');
+    }
+
+    /**
+     * @Then there should still be only one shipping category with code :code
+     */
+    public function thereShouldStillBeOnlyOneShippingCategoryWith(string $code): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::same(count($this->client->getCollectionItemsWithValue('code', $code)), 1);
+    }
+
+    /**
+     * @Then this shipping category name should be :name
+     */
+    public function thisShippingCategoryNameShouldBe(string $name): void
+    {
+        Assert::true(
+            $this->client->hasValue('name', $name),
```

Comment:
It is a minor thing, but `$this->client->hasValue` sounds slightly misleading for me. Maybe something like `$this->client->responseHasValue` would be a better choice? Because it is not the value of client, I suppose.